Automattic/wp-calypso
# Them tiers: Refactor theme tier selectors

Related to:
- https://github.com/Automattic/wp-calypso/issues/85899

## Proposed Changes

* Created separate selectors for `isThemeAllowedOnSite` and `getThemeTierForTheme`.
* Deleted the old hook and selector.
* Updated references to use the new selectors.

## Testing Instructions



* Open the Calypso Live link
* Goto `/themes`
* Add the `?flags=themes/tiers' flag
* Check that all the badges and everything else works as expected.


No before and after screenshots provided since there are no visual changes.
## Pre-merge Checklist
